test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -164,6 +166,29 @@ public void ItPublishesFrameworkDependentWithoutRid(string args)
                      .And.HaveStdOutContaining("Hello World");
         }
 
+        [Theory]
+        [InlineData("--self-contained --no-self-contained")]
+        [InlineData("--self-contained=true --no-self-contained")]
+        public void ItFailsToPublishWithConflictingArgument(string args)
+        {
+            var testAppName = "MSBuildTestApp";
+            var rid = DotnetLegacyRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier();
+
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance($"PublishApp_{rid ?? "none"}_{args ?? "none"}")
+                .WithSourceFiles()
+                .WithRestoreFiles();
+
+            var testProjectDirectory = testInstance.Root;
+
+            new PublishCommand()
+                .WithRuntime(rid)
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute(args)
+                .Should().Fail()
+                    .And.HaveStdErr();
```

Comment:
@dsychin whoops, sorry I missed this from my earlier review.  Could we change `HaveStdErr` to: `HaveStdErrContaining(LocalizableStrings.SelfContainAndNoSelfContainedConflict)`?

That way the test verifies that the correct error message is being output.